lib/uriSpid.js

```diff
@@ -0,0 +1,127 @@
+// TODO the usage of global is a temporary workatround that needs to be refactored out
+
+'use strict';
+
+/**
+ * @module uri
+ * This module contains routines and functions for getting the endpoints for SPiD API
+ * servers or various endpoints that the user will be redirected to.
+ */
+
+import { parse, format } from 'url';
+import SDKError from './SDKError';
+import Uri from './uri';
+
+/** @private */
+const SERVER_HOSTNAMES = {
```

Comment:
I'm not very happy about duplicating this logic. Let's fine a way to reuse this code (or at least the `normalizeHostname()` method). Ideally that method should be called from the `Builder`'s constructor but it can get a list of key&values that defines the normalization logic.